### PR TITLE
Update documentation for the new Algo debugging feature

### PR DESCRIPTION
**Pull request by Jaybsoni**
**Context:**
This is a follow up to this [PR](https://github.com/PennyLaneAI/pennylane/pull/5789) which added detailed documentation for the new PennyLane Debugger feature.

### CODE REVIEW

File: doc/introduction/inspecting_circuits.rst
Comment (trbromley) [inline doc/introduction/inspecting_circuits.rst:1]: Nice!

File: doc/code/qml_debugging.rst [hunk 1]
```diff
@@ -1,11 +1,8 @@
 qml.debugging
 =============
+.. currentmodule:: pennylane.debugging
 
-.. automodapi:: pennylane.debugging
-    :no-heading:
-    :no-inherited-members:
-    :skip: PLDB
-    :skip: pldb_device_manager
+.. automodule:: pennylane.debugging
```
Comment (trbromley) [inline doc/code/qml_debugging.rst:5]: Note that the table is now no longer [showing up](https://xanaduai-pennylane--5894.com.readthedocs.build/en/5894/code/qml_debugging.html):
image

I still think it's a useful table.

Comment (Jaybsoni) [reply doc/code/qml_debugging.rst]: Added 👍🏼

File: doc/code/qml_debugging.rst [hunk 2]
```diff
@@ -1,8 +1,25 @@
...
+   :format: html
+
 qml.debugging
 =============
-.. currentmodule:: pennylane.debugging
 
-.. automodule:: pennylane.debugging
+This module contains functionality for debugging quantum programs on simulator devices.
+
+:html:`<div class="summary-table">`
```
Comment (trbromley) [inline doc/code/qml_debugging.rst:9]: Curious why we couldn't use the built-in sphinx ones?

**Review by trbromley — APPROVED** [review]

File: doc/code/qml_debugging.rst (hunk 2 above)
Comment (Jaybsoni) [reply doc/code/qml_debugging.rst:9]: The builtin one is a module level summary that links the functions in the table using the path through the module (i.e `qml.debugging.debug_expval()`). In order to use the top level links we have to customize it.

File: doc/introduction/inspecting_circuits.rst [hunk 3]
```diff
@@ -166,6 +166,97 @@ results.
 All snapshots are numbered with consecutive integers, and if no tag was provided,
 the number of a snapshot is used as a key in the output dictionary instead.
 
+Interactive Debugging on Simulators
+-----------------------------------
+
+Pennylane allows for more interactive debugging of quantum circuits in a programatic
```
Comment (soranjh) [inline doc/introduction/inspecting_circuits.rst:172]:
```suggestion
PennyLane allows for more interactive debugging of quantum circuits in a programmatic 
```

**Review by soranjh — APPROVED** [review]

**Comment by soranjh** [comment]
@Jaybsoni Looks good. Please make sure that you target the correct base branch.

**Pull request merged**